lib/metadata/TableSchema.ts

```diff
@@ -218,11 +218,13 @@ export default class TableSchema {
     }
 
     if (this.ns.dateTime.equals(column.datatype.base)) {
-      return this.factory.literal(parseDateTime(value, column.datatype.format, this.timezone).toISO()!, this.ns.dateTime)
+      const date = parseDateTime(value, column.datatype.format, this.timezone)
+      return this.factory.literal(date ? date.toISO()! : value, this.ns.dateTime)
     }
 
     if (column.datatype.base.value === this.ns.date.value) {
```

Comment:
why compare the values? a few lines above for `dateTime` you used `equals`

Reply:
Before both were `===`. I refactored one but left the other unchanged in earlier PR 😅